src/core/DeltaLog.ts

```diff
@@ -0,0 +1,146 @@
+//   logKeepPrevious?: boolean;
+// }
+
+export class DeltaLog extends Array<Delta>{
+
+  static kDeltaLogFilename = `deltaLog.json`;
+  static kDeltaLogFile = `cves/${DeltaLog.kDeltaLogFilename}`;
+  // _options: DeltaLogOptions;
+  // _fullpath: string =
+  // _deltas: Delta[] = [];
+
+  // ----- constructor and factory functions ----- ----- ----- ----- ----- ----- ----- ----- ----- 
+
+  constructor(/* options: DeltaLogOptions */) {
+    super();
+    // this._options = options;
+    // this._options.path = options.path || `.`;
+    // this._options.filename = options.filename || `./test/activities_recent.json`;
+    // // this._options.mode = options.mode || `prepend`;
+    // this._options.logAlways = options.logAlways || false;
+    // this._options.logKeepPrevious = options.logKeepPrevious || false;
+    // this._fullpath = `${this._options.path}/${this._options.filename}`;
+
+    // console.log(`DeltaLog constructor:  options=${JSON.stringify(this)}`)
+    // console.log(`options=`, this._options);
+    // if (this._options.logKeepPrevious) {
+    // this._deltas = DeltaLog.readFile(this._fullpath);
+    // }
+    // else {
+    //   // fs.unlinkSync(this._fullpath);
+    //   this.clearActivities();
+    // }
+  }
+
+  /** constructs a DeltaLog by reading in the deltaLog file 
+   *  @param pruneOlderThan optional ISO date, any items older than that date will
+   *    not be included in the resulting DeltaLog
+   *  @param relFilepath optional path to the logfile (defaults to cves/deltaLog.json)
+   *  
+  */
+  static fromLogFile(
+    relFilepath?: string,
+    pruneOlderThan?: IsoDateString,
+  ): DeltaLog {
+    if (!pruneOlderThan) {
+      const days = process.env.CVES_DEFAULT_DELTA_LOG_HISTORY_IN_DAYS ?? "30";
+      pruneOlderThan = new IsoDateString().daysAgo(parseInt(days));
+      console.log(`setting pruning date to ${pruneOlderThan}`);
+    }
+    if (!relFilepath) {
+      console.log(`setting logFile to ${DeltaLog.kDeltaLogFile}`);
+      relFilepath = DeltaLog.kDeltaLogFile;
+    }
+    const pruneOlderThanTicks = pruneOlderThan.toNumber()
+    let json = [];
+    if (fs.existsSync(relFilepath)) {
+      const str = fs.readFileSync(relFilepath, { encoding: 'utf8', flag: 'r' });
+      if (str.length > 0) {
+        json = JSON.parse(str);
+      }
+    }
+    let log = new DeltaLog();
+    json.forEach(ele => {
+      const fetchTime = new IsoDateString(ele['fetchTime']);
+      if (fetchTime.toNumber() >= pruneOlderThanTicks) {
+        log.push(new Delta(ele));
+      }
+    });
+
+    return log;
+  }
+
+
+  // ----- class functions ----- ----- ----- ----- ----- ----- ----- ----- ----- 
+
+  /**
+   * prepends a delta to log
+   * @param delta the Delta object to prepend
+   */
+  prepend(delta: Delta): void {
+    this.unshift(delta);
+  }
+
+  /** sorts the Deltas in place by date 
+   *  @param direction: "latestFirst" | "latestLast"
+  */
+  sortByFetchTme(direction: "latestFirst" | "latestLast"): DeltaLog {
```

Comment:
is the direction arg documented enough, or would the direction options count as magic strings and should be converted to Symbols/enums? low priority